Microsoft.HealthVault.Fhir/Transformers/PractitionerToHealthVault.cs

```diff
@@ -0,0 +1,136 @@
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Support;
+using Microsoft.HealthVault.Fhir.Constants;
+using Microsoft.HealthVault.ItemTypes;
+
+namespace Microsoft.HealthVault.Fhir.Transformers
+{
+    public static partial class PractitionerToHealthVault
+    {
+        public static ItemTypes.PersonItem ToHealthVault(this Hl7.Fhir.Model.Practitioner fhirPractitioner)
+        {
+            return fhirPractitioner.ToPerson();
+        }
+
+        private static ItemTypes.PersonItem ToPerson(this Hl7.Fhir.Model.Practitioner fhirPractitioner)
+        {
+            if (fhirPractitioner == null || fhirPractitioner.Name.IsNullOrEmpty())
+                return null;
+            
+            var practitionerName = fhirPractitioner.Name.First(); // Let's consider just the first item
+
+            //name
+            var person = new ItemTypes.PersonItem()
+            {
+                PersonType = new CodableValue("Provider", new CodedValue("1", "person-types")),
+                Name = new ItemTypes.Name()
+                {
+                    Last = practitionerName.Family,
+                    Suffix = practitionerName.Suffix.Any() ? new ItemTypes.CodableValue(practitionerName.Suffix.First()) : null,
+                    Title = practitionerName.Prefix.Any() ? new ItemTypes.CodableValue(practitionerName.Prefix.First()) : null,
+                    First = practitionerName.Given.FirstOrDefault() ?? string.Empty,
+                    Middle = practitionerName.Given.ElementAtOrDefault(1) ?? string.Empty,
+                },
+                ContactInformation = new ContactInfo()
+            };            
+            if (!string.IsNullOrEmpty(practitionerName.Text))
+            {
+                person.Name.Full = practitionerName.Text;
+            }
+                              
+
+            //address
+            if (!fhirPractitioner.Address.IsNullOrEmpty())
```

Comment:
Very similar code is in 'Transformers/PatientToContact.cs' - is there a way to abstract the code and reuse?

Reply:
Fixed; Added a new ToHealthVault extension for Address and the same is used in both Practitioner and PatientToContact transformation classes